libfreerdp/gdi/gdi.c

```diff
@@ -662,24 +715,29 @@ static void gdi_patblt(rdpContext* context, PATBLT_ORDER* patblt)
 		WLog_ERR(TAG,  "unimplemented brush style:%d", brush->style);
 	}
 
+out_error:
 	gdi_SetTextColor(gdi->drawing->hdc, originalColor);
+	return ret;
 }
 
-static void gdi_scrblt(rdpContext* context, SCRBLT_ORDER* scrblt)
+static BOOL gdi_scrblt(rdpContext* context, SCRBLT_ORDER* scrblt)
 {
 	rdpGdi* gdi = context->gdi;
 
-	gdi_BitBlt(gdi->drawing->hdc, scrblt->nLeftRect, scrblt->nTopRect,
+	if (gdi_BitBlt(gdi->drawing->hdc, scrblt->nLeftRect, scrblt->nTopRect,
 			scrblt->nWidth, scrblt->nHeight, gdi->primary->hdc,
-			scrblt->nXSrc, scrblt->nYSrc, gdi_rop3_code(scrblt->bRop));
+			scrblt->nXSrc, scrblt->nYSrc, gdi_rop3_code(scrblt->bRop)) != 1)
```

Comment:
afaik return value 0 means failure and anything else is success. if that's true then I'd suggest to change this to == 0